worldedit-core/src/main/java/com/sk89q/worldedit/command/SelectionCommands.java

```diff
@@ -504,6 +505,133 @@ private BlockVector3[] getChangesForEachDir(int amount, boolean onlyHorizontal,
+                }
+            }
+        }
+
+        // If anything was found in the first pass, then the remaining variables are guaranteed to be set
+        if (!found) {
+            throw new StopExecutionException(TranslatableComponent.of(
+                        "worldedit.trim.no-blocks"));
+        }
+
+        int maxY = minY;
+
+        outer: for (int y = max.getBlockY(); y > minY; y--) {
+            for (int x = min.getBlockX(); x <= max.getBlockX(); x++) {
+                for (int z = min.getBlockZ(); z <= max.getBlockZ(); z++) {
+                    BlockVector3 vec = BlockVector3.at(x, y, z);
+
+                    if ((skipContains || originalRegion.contains(vec)) && mask.test(vec)) {
+                        maxY = y;
+                        break outer;
+                    }
+                }
+            }
+        }
+
+        int minX = 0;
+
+        outer: for (int x = min.getBlockX(); x <= max.getBlockX(); x++) {
+            for (int z = min.getBlockZ(); z <= max.getBlockZ(); z++) {
+                for (int y = minY; y <= maxY; y++) {
+                    BlockVector3 vec = BlockVector3.at(x, y, z);
+
+                    if ((skipContains || originalRegion.contains(vec)) && mask.test(vec)) {
+                        minX = x;
+                        break outer;
+                    }
+                }
+            }
+        }
+
+        int maxX = minX;
+
+        outer: for (int x = max.getBlockX(); x > minX; x--) {
+            for (int z = min.getBlockZ(); z <= max.getBlockZ(); z++) {
+                for (int y = minY; y <= maxY; y++) {
+                    BlockVector3 vec = BlockVector3.at(x, y, z);
+
+                    if ((skipContains || originalRegion.contains(vec)) && mask.test(vec)) {
+                        maxX = x;
+                        break outer;
+                    }
+                }
+            }
+        }
+
+        int minZ = 0;
+
+        outer: for (int z = min.getBlockZ(); z <= max.getBlockZ(); z++) {
+            for (int x = minX; x <= maxX; x++) {
+                for (int y = minY; y <= maxY; y++) {
+                    BlockVector3 vec = BlockVector3.at(x, y, z);
+
+                    if ((skipContains || originalRegion.contains(vec)) && mask.test(vec)) {
+                        minZ = z;
+                        break outer;
+                    }
+                }
+            }
+        }
+
+        int maxZ = minZ;
+
+        outer: for (int z = max.getBlockZ(); z > minZ; z--) {
+            for (int x = minX; x <= maxX; x++) {
+                for (int y = minY; y <= maxY; y++) {
+                    BlockVector3 vec = BlockVector3.at(x, y, z);
+
+                    if ((skipContains || originalRegion.contains(vec)) && mask.test(vec)) {
+                        maxZ = z;
+                        break outer;
+                    }
+                }
+            }
+        }
+
+        session.setRegionSelector(world, new CuboidRegionSelector(
```

Comment:
This command should probably do something similar to https://github.com/EngineHub/WorldEdit/commit/ba94660e27a90f39d2c14e2593cacf5c4b00f8b1, to ensure that `//sel extend` is retained when using this command